src/components/ui4/scenes/HomeSceneUi4.tsx

```diff
@@ -71,6 +76,61 @@ export const HomeSceneUi4 = (props: Props) => {
       .catch(e => console.log(String(e)))
   }, [])
 
+  // Check for PromoCard2 from info server:
+  React.useEffect(() => {
+    fetchInfo(`v1/promoCards2/${config.appId ?? 'edge'}`)
+      .then(async res => {
+        const infoData = await res.json()
+        const infoPromoCards = asArray(asPromoCard2)(infoData)
+
+        const buildNumber = getBuildNumber()
+        const countryCode = await getCountryCodeByIp()
+        const osType = Platform.OS.toLowerCase()
+        const osVersion = Platform.OS === 'ios' ? Platform.Version : `api${Platform.Version}`
+        const version = getVersion()
+        const currentDate = new Date()
+
+        setPromos(
+          infoPromoCards.filter(infoPromoCard => {
+            const { appVersion, minBuildNum, maxBuildNum, exactBuildNum, countryCodes, excludeCountryCodes, osTypes, osVersions, startIsoDate, endIsoDate } =
+              infoPromoCard
+
+            const isExactVersion = (appVersion != null && appVersion === version) || (exactBuildNum != null && exactBuildNum === buildNumber)
+            const isMinVersion = minBuildNum == null || minBuildNum <= buildNumber
+            const isMaxVersion = maxBuildNum == null || maxBuildNum >= buildNumber
+            const isVersionMatch = isExactVersion || (isMinVersion && isMaxVersion)
```

Comment:
isVersionMatch should also be true if there is no version specified in the card.